base/settings/R/site_pft_link_settings.R

```diff
@@ -26,7 +27,7 @@ site.pft.link.settings <-function(settings){
     def.pfts <- settings$pfts %>% purrr::map('name') %>% unlist() %>% as.character()
     
     # Create a simple pft tag for the pfts in LUT that are not in the pft tag
-    pft.l <- LUT$pft [!(LUT$pft %in% def.pfts)]
+    pft.l <- LUT$pft [!(LUT$pft %in% def.pfts)] %>% trimws() %>% unique()
     
     new.pfts <-  pft.l%>%
       purrr::map(function(lut.pft) {
```

Comment:
Is the goal here to drop elements of `pft.l` that are not in `def.pfts`? As written, this code will replace all cases where the `if` statement is `FALSE` with `NULL` (per the behavior of `if`), which is not quite the same thing.

I would probably simplify this as:

```r
new.pfts <- pft.l %>%
  purrr::drop(~.x %in% def.pfts) %>%
  purrr::map(~list(name = as.character(.x), constants = 1)) %>%
  setNames(rep("pft", length(pft.l))
```

Or even something like:

```r
new.pfts <- pft.l[!pft.l %in% def.pfts]  %>%
  purrr::map(~list(name = as.character(.x), constants = 1)) %>%
  setNames(rep("pft", length(pft.l))
```